redisson/src/main/java/org/redisson/config/BaseMasterSlaveServersConfig.java

```diff
@@ -70,6 +70,8 @@
     private int subscriptionConnectionPoolSize = 50;
 
     private long dnsMonitoringInterval = 5000;
+
+    private boolean dnsMonitoringOnFailure = false;
```

Comment:
Please correct me if I wrong, but it works only if dnsMonitoringInterval != -1

Reply:
No, it's independent of the monitoring interval, see `SentinelConnectionManager:315`. I extracted the functionality of the DNS check into a general method `performSentinelDNSCheck`. The interval is still only used in the `scheduleSentinelDNSCheck` method which now calls the new method.

Reply:
I think this function should be included without option.

Reply:
So you would suggest to execute `performSentinelDNSCheck` when dnsMonitoringInterval is -1, or regardless of the specified interval to always perform the check when all connections fail?

Reply:
Yes, always perform the check when all connections fail

Reply:
Okay, removed the config option and changed it so that `performSentinelDNSCheck` always gets called when all connections fail.